src/kademlia/node_id.cpp

```diff
@@ -70,6 +71,25 @@ int distance_exp(node_id const& n1, node_id const& n2)
 	return (std::max)(159 - distance(n1, n2).count_leading_zeroes(), 0);
 }
 
+int min_distance_exp(node_id const& n1, std::vector<node_id> const& ids)
+{
+	// specialized for cases of 0, 1 and 2 for performance reasons
+	if (ids.size() == 0) return 0;
+	if (ids.size() == 1) return distance_exp(n1, ids[0]);
+	if (ids.size() == 2)
+		return std::min(distance_exp(n1, ids[0]), distance_exp(n1, ids[1]));
+
+	int min = std::numeric_limits<int>::max();
```

Comment:
I guess this passed in msvc-15.. traditionally msvc has had problems with code like this by introducing a min and max macro from windows.h. the traditional solution is to not use identifiers called min or max and to say (std::numeric_limits<int>::max)(). But I guess this works now, maybe I should start cleaning up all the other places where I use the parenthesis trick :P